test/Base_Tests/src/Data/Decimal_Spec.enso

```diff
@@ -0,0 +1,81 @@
+from Standard.Base import all
+import Standard.Base.Errors.Common.Arithmetic_Error
+import Standard.Base.Errors.Common.Incomparable_Values
+import Standard.Base.Errors.Common.Type_Error
+import Standard.Base.Errors.Illegal_Argument.Illegal_Argument
+
+from Standard.Base.Data.Numbers import Number_Parse_Error
+
+from Standard.Test import all
+
+polyglot java import org.enso.base.numeric.Decimal_Utils
+
+float_id : Float -> Float
+float_id f =
+    Decimal_Utils.floatId f
+
+add_specs suite_builder =
+    suite_builder.group "arithmetic" group_builder->
+        group_builder.specify "should be able to pass a Float to Java" <|
+            vs = []
+                + [92233720368547758080.0]
+                + [9223372036854777000.0]
+                + [9223372036854776001.0]
+                + [9223372036854776000.0]
+                + [9223372036854775808.0]
+                + [9223372036854775000.0]
+                + [9223372036854774000.0]
+                + [9223372036854700000.0]
+                + [9223372036854000000.0]
+                + [9223372000000000000.0]
+                + [922337203685470000.0]
+                + [92233720368547000.0]
+                + [9223372036854700.0]
+                + [922337203685470.0]
+                + [92233720368547.0]
+                + [9223372036854.0]
+                + [922337203685.0]
+                + [92233720368.0]
+                + [9223372036.0]
+                + [922337203.0]
+                + [92233720.0]
+                + [9223372.0]
+                + [922337.0]
+                + [92233.0]
+                + [9223.0]
+                + [922.0]
+                + [-92233720368547758080.0]
+                + [-9223372036854777000.0]
+                + [-9223372036854776001.0]
+                + [-9223372036854776000.0]
+                + [-9223372036854775808.0]
+                + [-9223372036854775000.0]
+                + [-9223372036854774000.0]
+                + [-9223372036854700000.0]
+                + [-9223372036854000000.0]
+                + [-9223372000000000000.0]
+                + [-922337203685470000.0]
+                + [-92233720368547000.0]
+                + [-9223372036854700.0]
+                + [-922337203685470.0]
+                + [-92233720368547.0]
+                + [-9223372036854.0]
+                + [-922337203685.0]
+                + [-92233720368.0]
+                + [-9223372036.0]
+                + [-922337203.0]
+                + [-92233720.0]
+                + [-9223372.0]
+                + [-922337.0]
+                + [-92233.0]
+                + [-9223.0]
+                + [-922.0]
+
+            vs.map v->
+                r = float_id v
+                IO.println "float_id "+v.to_text+" "+r.to_text
```

Comment:
@JaroslavTulach I am seeing some unusual behavior here. I wonder if you get the same results when running `Decimal_Spec.enso` locally?

This test simply passes a Float to a Java function that returns its argument. It seems like the argument is changed when it reaches the Java in some cases.

```
float_id 9.223372036854776E19 9.223372036854776E19
float_id 9.223372036854778E18 9.223372036854778E18
float_id 9.223372036854776E18 9.223372036854776E18
float_id 9.223372036854776E18 9.223372036854776E18
float_id 9.223372036854776E18 9.223372036854776E18
**float_id 9.223372036854775E18 NaN**
**float_id 9.223372036854774E18 NaN**
**float_id 9.2233720368547E18 NaN**
**float_id 9.223372036854E18 NaN**
**float_id 9.223372E18 NaN**
**float_id 9.2233720368547E17 5.148800401705924E-247**
**float_id 9.2233720368547E16 1.7265376942307606E-302**
**float_id 9.2233720368547E15 4.663754807431019E-308**
float_id 9.2233720368547E14 9.2233720368547E14
float_id 9.2233720368547E13 9.2233720368547E13
float_id 9.223372036854E12 9.223372036854E12
float_id 9.22337203685E11 9.22337203685E11
float_id 9.2233720368E10 9.2233720368E10
float_id 9.223372036E9 9.223372036E9
float_id 9.22337203E8 9.22337203E8
float_id 9.223372E7 9.223372E7
float_id 9223372.0 9223372.0
float_id 922337.0 922337.0
float_id 92233.0 92233.0
float_id 9223.0 9223.0
float_id 922.0 922.0
float_id -9.223372036854776E19 -9.223372036854776E19
float_id -9.223372036854778E18 -9.223372036854778E18
**float_id -9.223372036854776E18 -0.0**
**float_id -9.223372036854776E18 -0.0**
**float_id -9.223372036854776E18 -0.0**
**float_id -9.223372036854775E18 -5.06E-321**
**float_id -9.223372036854774E18 -1.012E-320**
**float_id -9.2233720368547E18 -3.74383E-319**
**float_id -9.223372036854E18 -3.8349E-318**
**float_id -9.223372E18 -1.8208678612E-313**
**float_id -9.2233720368547E17 -8.390303882377555E246**
**float_id -9.2233720368547E16 -2.6059089326772877E302**
**float_id -9.2233720368547E15 -8.772742498128177E307**
float_id -9.2233720368547E14 -9.2233720368547E14
float_id -9.2233720368547E13 -9.2233720368547E13
float_id -9.223372036854E12 -9.223372036854E12
float_id -9.22337203685E11 -9.22337203685E11
float_id -9.2233720368E10 -9.2233720368E10
float_id -9.223372036E9 -9.223372036E9
float_id -9.22337203E8 -9.22337203E8
float_id -9.223372E7 -9.223372E7
float_id -9223372.0 -9223372.0
float_id -922337.0 -922337.0
float_id -92233.0 -92233.0
float_id -9223.0 -9223.0
float_id -922.0 -922.0

```

Reply:
> I wonder if you get the same results when running `Decimal_Spec.enso` locally?

Yes, I do. I am seeing the same results.